packages/flutter/test/widgets/app_title_test.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+const Color kTitleColor = const Color(0xFF333333);
+const String kTitleString = 'Hello World';
+
+Future<Null> pumpApp(WidgetTester tester, { GenerateAppTitle onGenerateTitle }) async {
+  await tester.pumpWidget(
+    new WidgetsApp(
+      supportedLocales: <Locale>[
+        const Locale('en', 'US'),
+        const Locale('en', 'GB'),
+      ],
+      title: kTitleString,
+      color: kTitleColor,
+      onGenerateTitle: onGenerateTitle,
+      onGenerateRoute: (RouteSettings settings) {
+        return new PageRouteBuilder<Null>(
+          pageBuilder: (BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+            return new Container();
+          }
+        );
+      },
+    ),
+  );
+}
+
+void main() {
+  testWidgets('Specified title and color are used to build a Title', (WidgetTester tester) async {
+    await pumpApp(tester);
+    expect(tester.widget(find.byType(Title)).title, kTitleString);
+    expect(tester.widget(find.byType(Title)).color, kTitleColor);
+  });
+
+  testWidgets('onGenerateTitle handles changing locales', (WidgetTester tester) async {
+    String generateTitle(BuildContext context) {
+      return Localizations.localeOf(context).toString();
+    }
+
+    await pumpApp(tester, onGenerateTitle: generateTitle);
+    expect(tester.widget(find.byType(Title)).title, 'en_US');
+    expect(tester.widget(find.byType(Title)).color, kTitleColor);
```

Comment:
you should also do a negative test to make sure en_GB isn't present here (and vice versa below), otherwise the pathological case where we go crazy and include every title would pass. :-)

Reply:
This test verifies that there's only one Title and that the Title's title is what's expected. Not sure what else I could check?